duckduckgo/iOS
# Add bookmarks and quick action today extensions

Task/Issue URL: https://app.asana.com/0/361428290920652/866517273193956
Tech Design URL: N/A

**Description**:
Add a quick actions and bookmarks widget to the app

**Steps to test this PR**:

Before beginning setup network sniffing (e.g with Charles proxy) so you can watch pixels fire.

**TEST QUICK ACTIONS**
1. Run the app
1. Add the `quick actions` today extensions
1. Tap the `search bar` and note that the app launches into a new search. Ensure that the `mqe_s` pixel fires
1. Return to the extension.
1. Tap the `Bookmarks` button and note that the app launches into the bookmarks screen. Ensure the `mqe_b` pixel fires.
1. Return to the extension.
1. Tap the `Clear Data` button and note that the app launches into the fire screen. Ensure the `mqe_f` pixel fires.

**TEST BOOKMARKS EXTENSION**
1. Run the app and ensure there are no bookmarks
1. Add the `bookmarks` today extension
1. Note the `No bookmarks yet` text
1. Tap `New private search` and note that the app launches into a new search. Ensure that the `mbe_s` pixel fires
1. Add some bookmarks into the app
1. Return to the widget and note that these now show. Tap a bookmark and ensure the correct one opens and the `mbe_b` pixel fires.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
